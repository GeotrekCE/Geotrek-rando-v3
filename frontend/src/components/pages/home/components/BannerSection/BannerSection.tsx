import { FormattedMessage } from 'react-intl';
import styled from 'styled-components';
import { colorPalette } from 'stylesheet';
import { BannerCarousel } from '../BannerCarousel/BannerCarousel';

const WelcomeText = styled.span`
  text-shadow: 0 0 20px ${colorPalette.home.shadowOnImages};
`;

interface BannerSectionProps {
  backgroundSourceUrl: string | string[];
  shouldDisplayText: boolean;
  type: 'image' | 'video' | 'carousel';
}

export const BannerSection: React.FC<BannerSectionProps> = ({
  backgroundSourceUrl,
  shouldDisplayText,
  type,
}) => {
  return (
    <div className="relative">
      {type === 'image' && typeof backgroundSourceUrl === 'string' && (
        <img
          src={backgroundSourceUrl}
          className="object-cover object-top overflow-hidden h-bannerSectionMobile desktop:h-bannerSectionDesktop w-full"
          data-testid="image"
        />
      )}
      {type === 'video' && typeof backgroundSourceUrl === 'string' && (
        <video
          autoPlay
          muted
          loop
          src={backgroundSourceUrl}
          className="object-cover object-center overflow-hidden h-bannerSectionMobile desktop:h-bannerSectionDesktop w-full"
          data-testid="video"
        />
      )}
      {type === 'carousel' && typeof backgroundSourceUrl === 'object' && (
        <BannerCarousel picturesUrl={backgroundSourceUrl} />
      )}
      <div className="absolute bottom-0 top-0 right-0 left-0 bg-gradient-to-t from-gradientOnImages to-transparent" />
      {shouldDisplayText && (
        <WelcomeText
          data-testid="text"
          className="text-white font-bold text-Mobile-H1 desktop:text-H1 text-center desktop:leading-tight
          absolute bottom-20 desktop:bottom-40 left-10vw right-10vw"
        >
          <FormattedMessage id="home.welcome-text" />
        </WelcomeText>
      )}
    </div>
  );
};
