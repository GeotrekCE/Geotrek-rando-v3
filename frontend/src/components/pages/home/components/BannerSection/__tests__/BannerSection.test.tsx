import { render } from 'services/testing/reactTestingLibraryWrapper';
import { BannerSection } from '../BannerSection';

describe('BannerSection', () => {
  it('should well render with a video', () => {
    const component = render(<BannerSection shouldDisplayText videoUrl="test.mp4" />);
    const asset = component.queryByTestId('video');

    expect(asset).toBeTruthy();

    expect(component).toMatchSnapshot();
  });

  it('should well render with a carousel', () => {
    const component = render(
      <BannerSection shouldDisplayText carouselUrls={['test1.jpg', 'test2.jpg']} />,
    );
    const asset = component.queryByTestId('carousel');

    expect(asset).toBeTruthy();

    expect(component).toMatchSnapshot();
  });

  it('should well render with a single picture', () => {
    const component = render(<BannerSection shouldDisplayText pictureUrl="test0.jpg" />);
    const asset = component.queryByTestId('image');

    expect(asset).toBeTruthy();

    expect(component).toMatchSnapshot();
  });

  test.each`
    isTextDefined |
    ${true}
    ${false}
  `('should adapt text if shouldDisplayText is $isTextDefined', ({ isTextDefined }) => {
    const component = render(
      <BannerSection shouldDisplayText={isTextDefined} pictureUrl="image.jpg" />,
    );

    const foundText = component.queryByTestId('text');

    expect(!!foundText).toBe(isTextDefined);
  });
});
