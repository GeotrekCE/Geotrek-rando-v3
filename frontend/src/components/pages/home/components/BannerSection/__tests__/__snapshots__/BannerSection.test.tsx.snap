// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BannerSection should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="relative"
      >
        <img
          class="object-cover object-top overflow-hidden h-bannerSectionMobile desktop:h-bannerSectionDesktop w-full"
          data-testid="image"
          src="test.jpg"
        />
        <div
          class="absolute bottom-0 top-0 right-0 left-0 bg-gradient-to-t from-gradientOnImages to-transparent"
        />
        <span
          class="BannerSection__WelcomeText-opht63-0 dYtjim text-white font-bold text-Mobile-H1 desktop:text-H1 text-center desktop:leading-tight absolute bottom-20 desktop:bottom-40 left-10vw right-10vw"
          data-testid="text"
        >
          L’écoguide numérique et local qui présente mes découvertes du territoire !
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="relative"
    >
      <img
        class="object-cover object-top overflow-hidden h-bannerSectionMobile desktop:h-bannerSectionDesktop w-full"
        data-testid="image"
        src="test.jpg"
      />
      <div
        class="absolute bottom-0 top-0 right-0 left-0 bg-gradient-to-t from-gradientOnImages to-transparent"
      />
      <span
        class="BannerSection__WelcomeText-opht63-0 dYtjim text-white font-bold text-Mobile-H1 desktop:text-H1 text-center desktop:leading-tight absolute bottom-20 desktop:bottom-40 left-10vw right-10vw"
        data-testid="text"
      >
        L’écoguide numérique et local qui présente mes découvertes du territoire !
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
